### DISCUSSION TITLE
Entity in Domainmodel violates Onion principle?

### DISCUSSION THREAD

**Discussion by HugoAusS** [dotnet/eShop#143] [Q&A]
The Order domain class derives from Entity which belongs to the entity framework namespace.
If I would like to keep my design fully following the onion principle would the reference to the EF not violate the principle, as EF is for DB access and in the core / domain I should not know anything about the persistence.  Or do I misunderstand it?

**Comment by jflaga** [comment]
Hi. I took a peek and noticed that the parent Entity class is not coming from the entity framework namespace, but from `eShop.Ordering.Domain.Seedwork` which is also located in the Domain module:

![image](https://github.com/dotnet/eShop/assets/108422823/4fc97529-a778-4b8d-a98f-126afc6b7f90)
